andela/metis-ah
# #159987410 User Notifications

#### What does this PR do?
Implement notifications

#### Description of Task to be completed?

- modify database relationship
- implement the functionality to create notifications
- implement the functionality to push notifications
- implement the functionality to view notifications and mark as read
#### How should this be manually tested?

- Clone the repository.
- Check out the `ft-notifications-159987410`  branch.
- Run `npm install` on your local machine.
- Run the db migrations using `npm run migrate`.
- Start the server using `npm run start_dev`.
- Using postman or any similar. get  a token upon registration and use it to test these routes:
**GET** `localhost:8000/api/v1/users/notifications` to get all notifications
**GET** `localhost:8000/api/v1/users/notifications/:notifyId` to get a single notification
**PUT** `localhost:8000/api/v1/users/notifications` to mark all as read
**PUT** `localhost:8000/api/v1/users/notifications/:notifyId` to mark one as read

#### Any background context you want to provide?
This feature will enable a user to get, view and mark notifications as read

#### What are the relevant pivotal tracker stories?
 #159987410
